src/app.js

```diff
@@ -0,0 +1,22 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+import asyncHandler from 'express-async-handler';
+
+import notFound from './Middleware/notFound';
+import errorHandler from './Middleware/errorHandler';
+
+const app = express();
+
+// We'll happily parse either JSON or URL-encoded bodies:
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+
+// Routes:
+// ...
+
+// Attach terminal "not found" & error handler
+// middleware for when things go awry:
+app.use(notFound);
+app.use(errorHandler);
```

Comment:
I'm less familiar with Express, but I take it the ordering of the `app.use()` statements is important here? Like the `notFound` that can toss an exception is later caught by the `errorHandler()` below in the logical ordering? Or does it not matter?

Reply:
The `notFound` middleware has to be a "terminal" route since it is handling the case where nothing else matches (and therefore `req.route` is null). The error handler could go wherever, but seemed nice to group with it.